contracts/IIdentityCloneFactory.sol

```diff
@@ -0,0 +1,33 @@
+// SPDX-License-Identifier: Apache-2.0
+pragma solidity >=0.8.0 <0.9.0;
+
+/**
+ * @dev DSNP Identity Factory Interface for creating identities via [EIP 1167](https://eips.ethereum.org/EIPS/eip-1167)
+ */
+interface IIdentityCloneFactory {
+    /**
+     * @dev event to log the created proxy contract address
+     */
+    event ProxyCreated(address addr);
+
+    /**
+     * @dev Creates a new identity with the message sender as the owner
+     * @dev [EIP 1167](https://eips.ethereum.org/EIPS/eip-1167) Proxy
+     * @param logic The address to use for the logic contract
+     *
+     * @dev This MUST emit ProxyCreated with the address of the new proxy contract
+     * @return The address of the newly created proxy contract
+     */
+    function createCloneProxy(address logic) external returns (address);
+
+    /**
+     * @dev Creates a new identity with the ecrecover address as the owner
+     * @dev [EIP 1167](https://eips.ethereum.org/EIPS/eip-1167) Proxy
+     * @param logic The address to use for the logic contract
+     * @param owner The initial owner's address of the new contract
+     *
+     * @dev This MUST emit ProxyCreated with the address of the new proxy contract
+     * @return The address of the newly created proxy contract
+     */
+    function createCloneProxyFor(address logic, address owner) external returns (address);
```

Comment:
Non-blocking: I wonder if the `owner` argument should come first since the function name ends with "For." This sort of reads like the clone proxy is for the logic address as it is now. Is there any solidity convention for prepositions in function names?

Reply:
In my head I read: create a clone proxy pointing at logic for owner thus the order, but I don't know of any particular conventions. Solidity param order is mostly determined by packing into the 256 byte blocks due to gas cost (which doesn't matter here with two 160 byte values).

Reply:
We could go for a multi-method, but I tend to shy away from those as they are not as easy to interact with in JS.

`createCloneProxyWithOwner`? (This will be a spec change first if we do it)

Reply:
`createCloneProxyWithOwner` sounds good to me